src/ENCRYPTO_utils/sndthread.cpp

```diff
@@ -118,6 +130,9 @@ void SndThread::ThreadMain() {
 				//delete sndlock;
 				run = false;
 			}
+			if(task->callback != NULL) {
```

Comment:
I think we should change NULL to nullptr here (and at some other places in this PR)